From 3e1c13993313a412119c10dc35c3a1d6a2a0b9b6 Mon Sep 17 00:00:00 2001
From: "yu.xie" <yu.xie@amlogic.com>
Date: Sat, 6 May 2023 11:26:56 +0000
Subject: [PATCH 1/3] v4l: CB2 Case jpeg -> nv21 test failed.. [1/1]

PD#SWPL-119395

Problem:
EOS send too much times in MJPEG.

Solution:
Fix the problem of EOS flag in MJPEG.

Verify:
AC200

Change-Id: Iafbf0d0617876ca66092e6c342cbfa714f75af71
Signed-off-by: yu.xie <yu.xie@amlogic.com>
---
 drivers/frame_provider/decoder_v4l/mjpeg/vmjpeg_multi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/frame_provider/decoder_v4l/mjpeg/vmjpeg_multi.c b/drivers/frame_provider/decoder_v4l/mjpeg/vmjpeg_multi.c
index 40638f74c..eafbe260f 100644
--- a/drivers/frame_provider/decoder_v4l/mjpeg/vmjpeg_multi.c
+++ b/drivers/frame_provider/decoder_v4l/mjpeg/vmjpeg_multi.c
@@ -1585,7 +1585,7 @@ static int notify_v4l_eos(struct vdec_s *vdec)
 	vdec_tracing(&ctx->vtr, VTRACE_DEC_PIC_0, aml_buf->index);
 	aml_buf_done(&ctx->bm, aml_buf, BUF_USER_DEC);
 
-	hw->eos = false;
+	hw->eos = true;
 
 	pr_info("[%d] mjpeg EOS notify.\n", ctx->id);
 
-- 
2.25.1

